azure-pipelines.yml

```diff
@@ -749,11 +749,23 @@ jobs:
 
 - job: bazel_test
   displayName: Bazel Software Build and Test
+  timeoutInMinutes: 120
   dependsOn: lint
-  pool:
+  pool: 
     vmImage: ubuntu-18.04
+  variables:
+  - name: bazelCacheGcpKeyPath
+    value: ''
   steps:
     - template: ci/install-package-dependencies.yml
+    - task: DownloadSecureFile@1
+      condition: eq(variables['Build.SourceBranchName'], 'master')
```

Comment:
Does this limit caching to only be written when changes are merged in and rolled up?

Reply:
This only limits writing to the cache to the master branch. The cache is publicly read-only, so all branches and PRs can read from the cache, but the GCP Key can only be downloaded by master.

Reply:
I guess that sounds prudent. I was hoping to empower and trust bazel and the VMs to write the cache locally based on bazel build results of untrusted code, but I guess configuring the VM to handle a secret without review would open a security hole. Some of that is going to require remote builds I guess.